### PR TITLE
CDAP-13558 Support CDC pipelines on Spark version 2.x

### PR DESCRIPTION
**Pull request by mlozbin-cybervisiontech**
JIRA - https://issues.cask.co/browse/CDAP-13558
WIKI - https://wiki.cask.co/display/CE/CDC+Solution+Enhancements

In scope of this PR:
- migrated to Java 8
- fixed dependency conflict with old cdap-spark-core (worked fine for maven but caused errors in IDE)
- removed too verbose logs
- minor refactoring to make code look similar

### CODE REVIEW

**Comment by mlozbin-cybervisiontech** [comment]
@albertshau, could you please take a look.

File: src/main/java/co/cask/cdc/plugins/common/SparkUtil.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import org.apache.spark.api.java.JavaRDD;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Utility methods for dealing with Spark configuration and data.
+ */
+public class SparkUtil {
```
Comment (albertshau) [inline src/main/java/co/cask/cdc/plugins/common/SparkUtil.java]: nit: Lets call this SparkConfigs. We usually name utility classes as some plural category that describes what's in the class. This is to try and prevent ever growing utility classes where everything in it is unrelated.

File: src/main/java/co/cask/cdc/plugins/source/oracle/BinaryMessageUtil.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import co.cask.cdap.api.common.Bytes;
+
+import javax.annotation.Nonnull;
+import java.nio.ByteBuffer;
+
+/**
+ * Utility methods for dealing with Spark configuration and data.
+ */
+public class BinaryMessageUtil {
```
Comment (albertshau) [inline src/main/java/co/cask/cdc/plugins/source/oracle/BinaryMessageUtil.java]: nit: let's name it BinaryMessages

File: src/main/java/co/cask/cdc/plugins/source/oracle/GoldenGateKafka.java [hunk 3]
```diff
@@ -27,37 +27,27 @@
...
 import java.nio.charset.StandardCharsets;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Iterator;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
+
+import static co.cask.cdc.plugins.source.oracle.BinaryMessageUtil.getBytesFromBinaryMessage;
```
Comment (albertshau) [inline src/main/java/co/cask/cdc/plugins/source/oracle/GoldenGateKafka.java]: style nit: our projects don't normally use use static imports. Instead, `BinaryMessageUtil.getBytesFromBinaryMessage()` is used in the code.

File: src/main/java/co/cask/cdc/plugins/source/sqlserver/CTInputDStream.java [hunk 4]
```diff
@@ -180,24 +162,19 @@ private long getCurrentTrackingVersion(Connection connection) throws SQLExceptio
...
 
-  private List<TableInformation> getCTEnabledTables() {
-    Connection connection = dbConnection.apply();
+  private List<TableInformation> getCTEnabledTables(Connection connection) throws SQLException {
     List<TableInformation> tableInformations = new LinkedList<>();
     String stmt = "SELECT s.name as schema_name, t.name AS table_name, ctt.* FROM sys.change_tracking_tables ctt " +
       "INNER JOIN sys.tables t on t.object_id = ctt.object_id INNER JOIN sys.schemas s on s.schema_id = t.schema_id";
-    try {
-      ResultSet rs = connection.createStatement().executeQuery(stmt);
+    try(ResultSet rs = connection.createStatement().executeQuery(stmt)) {
```
Comment (albertshau) [inline src/main/java/co/cask/cdc/plugins/source/sqlserver/CTInputDStream.java]: nit: add space between 'try' and '('

Comment (albertshau) [reply src/main/java/co/cask/cdc/plugins/source/sqlserver/CTInputDStream.java]: I believe this is supposed to be caught by our checkstyle. It looks like this project doesn't have checkstyle setup though. Can you add it in a separate PR (See https://github.com/data-integrations/google-cloud for an example)

**Review by albertshau — APPROVED** [review]
This looks good, only minor nits. If you fix and squash commits I'll merge.

**Comment by mlozbin-cybervisiontech** [comment]
@albertshau, addressed all comments and squashed.

**Pull request merged**